Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MetsXmlElementAccess.java

```diff
@@ -452,19 +453,21 @@ private StructLink createStructLink(LinkedList<Pair<String, String>> smLinkData)
     private Map<FileType, String> createFileUseByFileCache(Mets mets) {
         HashMap<FileType, String> fileUseMap = new HashMap<>();
         FileSec fileSec = mets.getFileSec();
+        Set<String> usedFileId = new HashSet<>();
+        boolean strictCheck = KitodoConfig.getConfig().getBoolean("useStrictMetsFileIdCheck", false);
         if (Objects.nonNull(fileSec)) {
             for (FileGrp fileGrp : fileSec.getFileGrp()) {
                 String use = fileGrp.getUSE();
                 for (FileType file : fileGrp.getFile()) {
-                    if (fileUseMap.containsKey(file)) {
+                    if (strictCheck && usedFileId.contains(file.getID())) {
                         throw new IllegalArgumentException(
                             "Corrupt file: file with id " + file.getID() + " is part of multiple groups"
```

Comment:
Is this still the correct error message for the new condition? With the updated check the exception would also be thrown if two files in the same file group use the same file ID, wouldn't it? Of course, that case is probably illegal as well, but does not match the exception message then.

Wouldn't it be better to have _two_ checks here, then? E.g. the original one that checks whether the current file is already in `fileUseMap` and a new one, that contains your added checks and throws an exception with a different message?

Reply:
1. You are right the exception message could be updated. Did you have a proposal?
2. The old test `fileUseMap.containsKey(file)` did not work as the compared object `file` is different as it's internal hash value differ between same data. So this test can't be restored.

Reply:
> 1. You are right the exception message could be updated. Did you have a proposal?

How about something like
`"Corrupt file: each METS file ID has to be unique but " + file.getID() + " is used multiple times!"`
?